GitHubLabels.ps1

```diff
@@ -565,6 +628,121 @@ function Set-GitHubLabel
+        If not supplied here, the DefaultRepositoryName configuration property value will be used.
+
+    .PARAMETER Uri
+        Uri for the repository.
+        The OwnerName and RepositoryName will be extracted from here instead of needing to provide
+        them individually.
+
+    .PARAMETER Issue
+        Issue number to add the label to.
+
+    .PARAMETER LabelName
+        Array of label names to add to the issue
+
+    .PARAMETER Replace
+        If supplied, will replace all of the labels on the issue with the provided labels.
+
+    .PARAMETER AccessToken
+        If provided, this will be used as the AccessToken for authentication with the
+        REST Api.  Otherwise, will attempt to use the configured value or will run unauthenticated.
+
+    .PARAMETER NoStatus
+        If this switch is specified, long-running commands will run on the main thread
+        with no commandline status update.  When not specified, those commands run in
+        the background, enabling the command prompt to provide status information.
+        If not supplied here, the DefaultNoStatus configuration property value will be used.
+
+    .EXAMPLE
+        Add-GitHubLabel -OwnerName PowerShell -RepositoryName PowerShellForGitHub -Issue 1 -LabelName $labels
+
+        Adds labels to an issue in the PowerShellForGitHub project.
+#>
+    [CmdletBinding(
+        SupportsShouldProcess,
+        DefaultParametersetName='Elements')]
+    [Diagnostics.CodeAnalysis.SuppressMessageAttribute("PSShouldProcess", "", Justification="Methods called within here make use of PSShouldProcess, and the switch is passed on to them inherently.")]
+    param(
+        [Parameter(ParameterSetName='Elements')]
+        [string] $OwnerName,
+
+        [Parameter(ParameterSetName='Elements')]
+        [string] $RepositoryName,
+
+        [Parameter(
+            Mandatory,
+            ParameterSetName='Uri')]
+        [string] $Uri,
+
+        [Parameter(Mandatory)]
+        [int] $Issue,
+
+        [Parameter(Mandatory)]
+        [string[]] $LabelName,
+
+        [switch] $Replace,
+
+        [string] $AccessToken,
+
+        [switch] $NoStatus
+    )
+
+    Write-InvocationLog
+
+    $elements = Resolve-RepositoryElements
+    $OwnerName = $elements.ownerName
+    $RepositoryName = $elements.repositoryName
+
+    $telemetryProperties = @{
+        'OwnerName' = (Get-PiiSafeString -PlainText $OwnerName)
+        'RepositoryName' = (Get-PiiSafeString -PlainText $RepositoryName)
+        'labelCount' = $LabelName.Count
+    }
+
+    $hashBody = @{
+        'labels' = $LabelName
+    }
+
+    if ($Replace)
+    {
+        $method = 'Put'
+    }
+    else
+    {
+        $method = 'Post'
+    }
+
+    $params = @{
+        'UriFragment' = "repos/$OwnerName/$RepositoryName/issues/$Issue/labels"
+        'Body' = (ConvertTo-Json -InputObject $hashBody)
+        'Method' = $method
+        'Description' =  "Adding label $Name to issue $Issue in $RepositoryName"
```

Comment:
This definitely can't be working given that you currently don't have a `$Name` parameter.  On top of that, it's not totally accurate if using the `-Replace` switch as currently designed, as it's _replacing_ the labels.

Reply:
If I change this to Set-GithubLabel we are going to run into the same ass with Remove-GithubLabel. Setting the title/color/etc of a github label is a lot different than setting the labels on an issue. I think at this point it would be better to have a separate set of Add-GithubIssueLabel/Remove-GithubIssueLabel/Set-GithubIssueLabel commands just for issues. There will not be commands for Milestones since labels put on labels are not directly set on a Milestone but instead computed by all of the labels that are set on issues that are a part of a Milestone.

Reply:
Ok, I'm convinced.  Break 'em out that way.  Thanks!